libfwupdplugin/fu-plugin.c

```diff
@@ -451,6 +450,27 @@ fu_plugin_open (FuPlugin *self, const gchar *filename, GError **error)
 	return TRUE;
 }
 
+/* order of usefullness to the user */
+static const gchar *
+fu_plugin_build_device_update_error (FuPlugin *self)
+{
+	if (fu_plugin_has_flag (self, FWUPD_PLUGIN_FLAG_NO_HARDWARE))
+		return "Required hardware was not found";
+	if (fu_plugin_has_flag (self, FWUPD_PLUGIN_FLAG_LEGACY_BIOS))
+		return "Firmware can not be updated in legacy BIOS mode";
+	if (fu_plugin_has_flag (self, FWUPD_PLUGIN_FLAG_CAPSULES_UNSUPPORTED))
+		return "UEFI capsule updates not available or enabled";
+	if (fu_plugin_has_flag (self, FWUPD_PLUGIN_FLAG_UNLOCK_REQUIRED))
+		return "Firmware updates disabled; run 'fwupdmgr unlock' to enable";
```

Comment:
this won't really translate well into GUI clients

Reply:
True, what about just `A device requires unlocking`